decompiler/frontend/binaryninja/handlers/assignments.py

```diff
@@ -73,9 +73,9 @@ def lift_set_field(self, assignment: mediumlevelil.MediumLevelILSetVarField, is_
             struct_variable = self._lifter.lift(assignment.dest, is_aliased=True, parent=assignment)
             destination = MemberAccess(
                 offset=assignment.offset,
-                member_name=struct_variable.type.get_member_by_offset(assignment.offset).name,
+                member_name=struct_variable.type.get_member_name_by_offset(assignment.offset),
                 operands=[struct_variable],
-                writes_memory=assignment.ssa_memory_version,
+                writes_memory=assignment.ssa_memory_version + 1,
```

Comment:
As far as I understand it, `assignment.ssa_memory_version` returns the memory version used during the assignment, i.e. before the assignment took place. Therefore the assignment creates the memory version `assignment.ssa_memory_version + 1`. Without this change, later stages would sometimes crash or malfunction.
Instead of `assignment.ssa_memory_version + 1` it would also be possible to use `assignment.dest.version`. In the cases I observed the resulting value would be identical. However, I am unsure if  `assignment.dest.version` will *always* refer to the memory version in set field instructions.
To me, it was unclear which version is better, so I just picked the former one. Do you have a preference?

Reply:
I kinda feel like `assignment.dest.version` would let me sleep better at night - sometimes we got a single line updating the memory version twice